gateway/mock_response.go

```diff
@@ -57,3 +77,77 @@ func mockFromConfig(tykMockRespOp *oas.MockResponse) (int, []byte, map[string]st
 
 	return code, body, headers
 }
+
+func mockFromOAS(r *http.Request, operation *openapi3.Operation, fromOASExamples *oas.FromOASExamples) (int, string, []byte, map[string]string, error) {
+	exampleName := fromOASExamples.ExampleName
+	if headerExampleName := r.Header.Get(acceptExampleName); headerExampleName != "" {
+		exampleName = headerExampleName
+	}
+
+	code := 200
+	if fromOASExamples.Code != 0 {
+		code = fromOASExamples.Code
+	}
+
+	var err error
+	if headerCode := r.Header.Get(acceptCode); headerCode != "" {
+		if code, err = strconv.Atoi(headerCode); err != nil {
+			return http.StatusBadRequest, "", nil, nil, fmt.Errorf("given code %s is not a valid integer value", headerCode)
+		}
+	}
+
+	contentType := "application/json"
+	if fromOASExamples.ContentType != "" {
+		contentType = fromOASExamples.ContentType
+	}
+
+	if headerContentType := r.Header.Get(acceptContentType); headerContentType != "*/*" && headerContentType != "" {
+		contentType = headerContentType
+	}
+
+	response, ok := operation.Responses[strconv.Itoa(code)]
+	if !ok {
+		return http.StatusNotFound, "", nil, nil, fmt.Errorf("there is no example response for the code: %d", code)
+	}
+
+	media := response.Value.Content.Get(contentType)
+	if media == nil {
+		return http.StatusNotFound, "", nil, nil, errors.New("there is no example response for the content type: " + contentType)
+	}
+
+	headers := make(map[string]string)
+	/*for key, val := range response.Value.Headers {
```

Comment:
?

Reply:
These comments will be active when I push extractor code.